discord/poll.py

```diff
@@ -0,0 +1,686 @@
+        return payload  # type: ignore
+
+    @classmethod
+    def from_dict(cls, *, data: PollMediaPayload) -> PollMedia:
+        """Returns a new instance of this class from a payload."""
+
+        emoji = data.get('emoji')
+
+        if emoji:
+            return cls(text=data['text'], emoji=PartialEmoji.from_dict(emoji))
+        return cls(text=data['text'])
+
+
+class PollAnswerBase:
+    def __init__(self, *, id: int, message: Optional[Message], state: Optional[ConnectionState]) -> None:
+        self.id: int = id
+        self._message: Optional[Message] = message
+        self._state: Optional[ConnectionState] = state
+
+    def _update(self, message: Message) -> None:
+        self._message = message
+        self._state = message._state
+
+    async def users(
+        self, *, limit: Optional[int] = None, after: Optional[Snowflake] = None
+    ) -> AsyncIterator[Union[User, Member]]:
+        """Returns an :term:`asynchronous iterator` representing the users that have voted to this answer.
+
+        The ``after`` parameter must represent a user
+        and meet the :class:`abc.Snowflake` abc.
+
+        .. warning::
+
+            This can only be called when the poll is accessed via :attr:`Message.poll`.
+
+        Examples
+        --------
+
+        Usage ::
+
+            async for user in poll_answer.users():
+                print(f'{voter} has voted for {poll_answer}!')
+
+        Flattening into a list: ::
+
+            users = [user async for user in poll_answer.users()]
+            # users is now a list of User
+
+        Parameters
+        ----------
+        limit: Optional[:class:`int`]
+            The max number of users to return. Can be up to 100.
+        after: Optional[:class:`abc.Snowflake`]
+            Fetches users after this ID, or every user if ``None``.
+
+        Raises
+        ------
+        HTTPException
+            Retrieving the users failed.
+
+        Yields
+        ------
+        Union[:class:`User`, :class:`Member`]
+            The users that voted for this poll answer. This can be a :class:`Member` object if the poll
+            is in a guild-message context, for other contexts it will always return a :class:`User`, or when
+            the member left the guild.
+        """
+
+        if not self._message or not self._state:  # Make type checker happy
+            raise RuntimeError('You cannot fetch users in a non-message-attached poll')
+
+        data = await self._state.http.get_poll_answer_voters(
+            self._message.channel.id, self._message.id, self.id, after.id if after else None, limit
+        )
+
+        if not self._message.guild:
+            for user in data.get('users'):
+                yield User(state=self._state, data=user)
+
+        else:
+            guild = self._message.guild
+
+            for user in data.get('users'):
+                member = guild.get_member(int(user['id']))
+
+                yield member or User(state=self._state, data=user)
```

Comment:
This needs to be reworked entirely, if the limit is >100 we should instead keep hitting the endpoint until it is exhaused, reaction users does the same thing.

Additionally, I believe this should be called `voters`